.github/workflows/main.yml

```diff
@@ -0,0 +1,116 @@
+              flavor: centos
+              major: 7
+            java_major: 8
+            jdist: jdk
+          - tomcat_major: 8
+            base_image:
+              flavor: centos
+              major: 7
+            java_major: 11
+            jdist: jdk
+    runs-on: ubuntu-latest
+    steps:
+      - id: co
+        name: checkout project
+        uses: actions/checkout@v2
+      - id: vars
+        name: compute image tag
+        env:
+          IMAGE_BASE_NAME: tomcat${{ matrix.tomcat_major }}-${{ matrix.jdist }}${{ matrix.java_major }}-${{ matrix.base_image.flavor }}${{ matrix.base_image.major }}
+        run: |
+          if [[ "${{ github.ref_name }}" != "master" ]]
+          then
+            echo "::set-output name=image_tag::${IMAGE_BASE_NAME}-${GITHUB_REF_NAME//\//-}"
+          elif [[ "${{ github.event.head_commit.message }}" =~ .*\[release\].* ]]
+          then
+            echo "::set-output name=image_tag::$IMAGE_BASE_NAME"
+          else
+            echo "::set-output name=image_tag::${IMAGE_BASE_NAME}-release-candidate"
+          fi
+          echo "::set-output name=image_created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
+          echo "::set-output name=image_anchor::$(date -u +'%Y%m%d%H%M')"
+      - id: q_login
+        name: Login to Quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - id: dh_login
+        if: contains(github.event.head_commit.message, '[release]') && github.ref_name == 'master'
+        name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - id: q_push
+        name: Build and push to Quay.io
+        uses: docker/build-push-action@v2.8.0
+        with:
+          push: true
+          build-args: |
+            no-cache=true
+            JDIST=${{ matrix.jdist }}
+            DISTRIB_NAME=${{ matrix.base_image.flavor }}
+            DISTRIB_MAJOR=${{ matrix.base_image.major }}
+            JAVA_MAJOR=${{ matrix.java_major }}
+            TOMCAT_MAJOR=${{ matrix.tomcat_major }}
+            REVISION=${{ github.run_number }}
+            CREATED=${{ steps.vars.outputs.image_created }}
+          tags: |
+            quay.io/${{ env.IMAGE_REGISTRY_NAMESPACE }}/${{ env.IMAGE_REPOSITORY }}:${{ steps.vars.outputs.image_tag }}-${{ steps.vars.outputs.image_anchor }}
+            quay.io/${{ env.IMAGE_REGISTRY_NAMESPACE }}/${{ env.IMAGE_REPOSITORY }}:${{ steps.vars.outputs.image_tag }}
+          target: TOMCAT_BASE_IMAGE
+      - id: dh_push
+        if: contains(github.event.head_commit.message, '[release]') && github.ref_name == 'master'
+        name: Build and push to Docker Hub
+        uses: docker/build-push-action@v2.8.0
+        with:
+          push: true
+          build-args: |
+            no-cache=true
```

Comment:
are you forcing a full rebuild when also releasing on dockerhub? why?

Reply:
not particularly. Just reusing the --no-cache flag that was already used in previous build, but I have to say that it sounds safer to use it given the dynamic nature of this builds

Reply:
and I see that builds run in a couple of minutes so nevermind